jupyterlab_bigquery/src/components/query_editor/query_text_editor/query_text_editor.tsx

```diff
@@ -36,55 +60,210 @@ class QueryTextEditor extends React.Component<
   }
 
   handleEditorDidMount(_, editor) {
     this.editor = editor;
+
+    this.editor.onKeyUp(() => {
+      // eslint-disable-next-line no-extra-boolean-cast
+      if (!!this.timeoutAlarm) {
+        clearTimeout(this.timeoutAlarm);
+        this.setState({ errorMsg: null });
+      }
+      this.timeoutAlarm = setTimeout(this.checkSQL.bind(this), 1500);
+    });
+  }
+
+  checkSQL() {
+    const query = this.editor.getValue();
+    this.pagedQueryService.request(
+      { query, jobConfig: {}, dryRunOnly: true },
+      (state, _, response) => {
+        if (state === JobState.Fail) {
+          this.setState({
+            errorMsg: response as string,
+          });
+        }
+      }
+    );
+  }
+
+  readableBytes(bytes: number) {
+    const i = Math.floor(Math.log(bytes) / Math.log(1024));
+    const sizes = ['B', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
+
+    return (bytes / Math.pow(1024, i)).toFixed(2) + ' ' + sizes[i];
+  }
+
+  renderButton() {
+    const buttonState = this.state.buttonState;
+    let color = undefined;
+    let content = undefined;
+
+    switch (buttonState) {
+      case ButtonStates.PENDING:
+        color = 'default';
+        content = (
+          <div style={{ display: 'flex', flexDirection: 'row', width: '100%' }}>
+            <CircularProgress size="75%" style={{ alignSelf: 'center' }} />
+            <Typography variant="button">Cancel</Typography>
+          </div>
+        );
+        break;
+      case ButtonStates.READY:
+        color = 'primary';
+        content = 'Submit';
+        break;
+      case ButtonStates.ERROR:
+        color = 'secondary';
+        content = 'Error';
+        break;
+    }
+
+    return (
+      <Button
+        color={color}
+        variant="contained"
+        onClick={this.handleButtonClick.bind(this)}
+        className={styleSheet.queryButton}
+      >
+        {content}
+      </Button>
+    );
+  }
+
+  renderOptionalText(text, config = {}) {
+    // eslint-disable-next-line no-extra-boolean-cast
+    if (!!text) {
+      return (
+        <Typography
+          style={{
+            marginRight: '10px',
+            marginLeft: '10px',
+            alignSelf: 'center',
+            justifySelf: 'center',
+          }}
+          variant="body1"
+          {...config}
+        >
+          {text}
+        </Typography>
+      );
+    }
+
+    return undefined;
   }
 
   render() {
+    // eslint-disable-next-line no-extra-boolean-cast
+    const readableSize = !!this.state.bytesProcessed
+      ? 'Processed ' + this.readableBytes(this.state.bytesProcessed)
+      : null;
+
+    const errMsg = this.state.errorMsg;
+
     return (
       <div style={{ height: '50%' }}>
         <Editor
           width="100vw"
           height="40vh"
           theme={'light'}
           language={'sql'}
-          value={'// type your code...'}
+          value={
+            'SELECT * FROM `jupyterlab-interns-sandbox.covid19_public_forecasts.county_14d` LIMIT 10'
```

Comment:
I'm pretty sure this just an example query, but we should probably preload with a query based on the context menu or Details page.